Occasional NPE thrown with the Maven resolver

```
java.lang.NullPointerException: at index 1
	at com.google.common.collect.ObjectArrays.checkElementNotNull(ObjectArrays.java:232)
	at com.google.common.collect.ObjectArrays.checkElementsNotNull(ObjectArrays.java:222)
	at com.google.common.collect.ObjectArrays.checkElementsNotNull(ObjectArrays.java:216)
	at com.google.common.collect.ImmutableList.construct(ImmutableList.java:354)
	at com.google.common.collect.ImmutableList.copyOf(ImmutableList.java:266)
	at com.github.bazelbuild.rules_jvm_external.resolver.maven.ErrorReportingListener.getExceptions(ErrorReportingListener.java:49)
	at com.github.bazelbuild.rules_jvm_external.resolver.maven.MavenResolver.resolve(MavenResolver.java:216)
	at com.github.bazelbuild.rules_jvm_external.resolver.cmd.Main.main(Main.java:74)
```

Retrying usually helps, so I am not sure of steps to reproduce.

Here's the full trace:

```
[Fatal Error] plexus-1.0.4.pom:150:34: The entity "oslash" was referenced, but not declared.
[ParallelDescriptorResolver-0-0] WARN org.eclipse.aether.internal.impl.DefaultRepositoryEventDispatcher - Failed to dispatch repository event to com.github.bazelbuild.rules_jvm_external.resolver.maven.CompoundListener
java.lang.RuntimeException: Unable to determine packaging
	at com.github.bazelbuild.rules_jvm_external.resolver.maven.CoordinateGatheringListener.artifactResolved(CoordinateGatheringListener.java:98)
	at com.github.bazelbuild.rules_jvm_external.resolver.maven.CompoundListener.lambda$artifactResolved$3(CompoundListener.java:46)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
	at com.github.bazelbuild.rules_jvm_external.resolver.maven.CompoundListener.artifactResolved(CompoundListener.java:46)
	at org.eclipse.aether.internal.impl.DefaultRepositoryEventDispatcher.dispatch(DefaultRepositoryEventDispatcher.java:120)
	at org.eclipse.aether.internal.impl.DefaultRepositoryEventDispatcher.dispatch(DefaultRepositoryEventDispatcher.java:88)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.artifactResolved(DefaultArtifactResolver.java:669)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.evaluateDownloads(DefaultArtifactResolver.java:641)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.performDownloads(DefaultArtifactResolver.java:545)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:449)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts(DefaultArtifactResolver.java:261)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifact(DefaultArtifactResolver.java:243)
	at org.apache.maven.repository.internal.DefaultModelResolver.resolveModel(DefaultModelResolver.java:158)
	at org.apache.maven.repository.internal.DefaultModelResolver.resolveModel(DefaultModelResolver.java:[204](https://gitlab.booking.com/bplatform/infra/-/jobs/210620324#L204))
	at org.apache.maven.model.building.DefaultModelBuilder.readParentExternally(DefaultModelBuilder.java:1009)
	at org.apache.maven.model.building.DefaultModelBuilder.readParent(DefaultModelBuilder.java:801)
	at org.apache.maven.model.building.DefaultModelBuilder.build(DefaultModelBuilder.java:327)
	at org.apache.maven.model.building.DefaultModelBuilder.build(DefaultModelBuilder.java:243)
	at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.loadPom(DefaultArtifactDescriptorReader.java:284)
	at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.readArtifactDescriptor(DefaultArtifactDescriptorReader.java:175)
	at org.eclipse.aether.internal.impl.collect.bf.BfDependencyCollector.resolveCachedArtifactDescriptor(BfDependencyCollector.java:464)
	at org.eclipse.aether.internal.impl.collect.bf.BfDependencyCollector.resolveDescriptorForVersion(BfDependencyCollector.java:450)
	at org.eclipse.aether.internal.impl.collect.bf.BfDependencyCollector.lambda$resolveArtifactDescriptorAsync$1(BfDependencyCollector.java:417)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658)
	at org.eclipse.aether.internal.impl.collect.bf.BfDependencyCollector.lambda$resolveArtifactDescriptorAsync$4(BfDependencyCollector.java:416)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: org.xml.sax.SAXParseException; systemId: file:/tmp/resolver-home829077590963521214/.m2/repository/org/codehaus/plexus/plexus/1.0.4/plexus-1.0.4.pom; lineNumber: 150; columnNumber: 34; The entity "oslash" was referenced, but not declared.
	at java.xml/com.sun.org.apache.xerces.internal.parsers.DOMParser.parse(DOMParser.java:261)
	at java.xml/com.sun.org.apache.xerces.internal.jaxp.DocumentBuilderImpl.parse(DocumentBuilderImpl.java:339)
	at java.xml/javax.xml.parsers.DocumentBuilder.parse(DocumentBuilder.java:[206](https://gitlab.booking.com/bplatform/infra/-/jobs/210620324#L206))
	at com.github.bazelbuild.rules_jvm_external.resolver.maven.CoordinateGatheringListener.artifactResolved(CoordinateGatheringListener.java:62)
	... 29 more
java.lang.NullPointerException: at index 1
	at com.google.common.collect.ObjectArrays.checkElementNotNull(ObjectArrays.java:232)
	at com.google.common.collect.ObjectArrays.checkElementsNotNull(ObjectArrays.java:222)
	at com.google.common.collect.ObjectArrays.checkElementsNotNull(ObjectArrays.java:216)
	at com.google.common.collect.ImmutableList.construct(ImmutableList.java:354)
	at com.google.common.collect.ImmutableList.copyOf(ImmutableList.java:266)
	at com.github.bazelbuild.rules_jvm_external.resolver.maven.ErrorReportingListener.getExceptions(ErrorReportingListener.java:49)
	at com.github.bazelbuild.rules_jvm_external.resolver.maven.MavenResolver.resolve(MavenResolver.java:216)
	at com.github.bazelbuild.rules_jvm_external.resolver.cmd.Main.main(Main.java:74)
```

The first part of the error is from https://github.com/bazelbuild/rules_jvm_external/issues/1144.

--- a/private/tools/java/com/github/bazelbuild/rules_jvm_external/resolver/maven/ErrorReportingListener.java
+++ b/private/tools/java/com/github/bazelbuild/rules_jvm_external/resolver/maven/ErrorReportingListener.java
@@ -15,6 +15,7 @@
 package com.github.bazelbuild.rules_jvm_external.resolver.maven;
 
 import com.google.common.collect.ImmutableList;
+import java.util.Collections;
 import java.util.LinkedList;
 import java.util.List;
 import org.eclipse.aether.AbstractRepositoryListener;
@@ -22,30 +23,33 @@ import org.eclipse.aether.RepositoryEvent;
 
 class ErrorReportingListener extends AbstractRepositoryListener {
 
-  private final List<Exception> exceptions = new LinkedList<>();
+  private final List<Exception> exceptions = Collections.synchronizedList(new LinkedList<>());
 
   @Override
   public void artifactDescriptorInvalid(RepositoryEvent event) {
-    if (event.getException() != null) {
-      exceptions.add(event.getException());
-    }
+    addExceptionIfPresent(event);
   }
 
   @Override
   public void artifactDescriptorMissing(RepositoryEvent event) {
-    if (event.getException() != null) {
-      exceptions.add(event.getException());
-    }
+    addExceptionIfPresent(event);
   }
 
   @Override
   public void metadataInvalid(RepositoryEvent event) {
-    if (event.getException() != null) {
+    addExceptionIfPresent(event);
+  }
+
+  private void addExceptionIfPresent(RepositoryEvent event) {
+    if (event != null && event.getException() != null) {
       exceptions.add(event.getException());
     }
   }
 
   public List<Exception> getExceptions() {
-    return ImmutableList.copyOf(exceptions);
+    // https://www.baeldung.com/java-synchronized-collections#the-synchronizedList-method
+    synchronized (exceptions) {
+      return ImmutableList.copyOf(exceptions);
+    }
   }
 }

--- a/tests/com/github/bazelbuild/rules_jvm_external/resolver/maven/BUILD
+++ b/tests/com/github/bazelbuild/rules_jvm_external/resolver/maven/BUILD
@@ -41,6 +41,24 @@ java_test(
 )
 
 java_test(
+    name = "ErrorReportingListenerTest",
+    size = "small",
+    srcs = ["ErrorReportingListenerTest.java"],
+    test_class = "com.github.bazelbuild.rules_jvm_external.resolver.maven.ErrorReportingListenerTest",
+    deps = [
+        "//private/tools/java/com/github/bazelbuild/rules_jvm_external/resolver/maven",
+        artifact(
+            "junit:junit",
+            repository_name = "regression_testing_coursier",
+        ),
+        artifact(
+            "org.apache.maven.resolver:maven-resolver-api",
+            repository_name = "rules_jvm_external_deps",
+        ),
+    ],
+)
+
+java_test(
     name = "MavenResolverTest",
     size = "medium",
     srcs = ["MavenResolverTest.java"],

--- a/tests/com/github/bazelbuild/rules_jvm_external/resolver/maven/ErrorReportingListenerTest.java
+++ b/tests/com/github/bazelbuild/rules_jvm_external/resolver/maven/ErrorReportingListenerTest.java
@@ -1,0 +1,49 @@
+// Copyright 2026 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.github.bazelbuild.rules_jvm_external.resolver.maven;
+
+import static org.junit.Assert.assertEquals;
+
+import org.eclipse.aether.DefaultRepositorySystemSession;
+import org.eclipse.aether.RepositoryEvent;
+import org.junit.Test;
+
+public class ErrorReportingListenerTest {
+
+  @Test
+  public void getExceptionsShouldBeThreadSafe() throws Exception {
+    ErrorReportingListener listener = new ErrorReportingListener();
+    int threadCount = 8;
+    int eventsPerThread = 500;
+    Thread[] threads = new Thread[threadCount];
+
+    for (int t = 0; t < threadCount; t++) {
+      threads[t] = new Thread(() -> {
+        for (int i = 0; i < eventsPerThread; i++) {
+          RepositoryEvent.Builder builder = new RepositoryEvent.Builder(
+              new DefaultRepositorySystemSession(),
+              RepositoryEvent.EventType.ARTIFACT_DESCRIPTOR_MISSING);
+          builder.setException(new RuntimeException("err"));
+          listener.artifactDescriptorMissing(builder.build());
+        }
+      });
+    }
+
+    for (Thread thread : threads) thread.start();
+    for (Thread thread : threads) thread.join();
+
+    assertEquals(threadCount * eventsPerThread, listener.getExceptions().size());
+  }
+}

